CMakeLists.txt

```diff
@@ -100,22 +100,29 @@ endif( )
 # BUILD_SHARED_LIBS is a cmake built-in; we make it an explicit option such that it shows in cmake-gui
 option( BUILD_SHARED_LIBS "Build hipBLAS as a shared library" ON )
 
-# Find CUDA if the user wants a CUDA version.
-option(USE_CUDA "Look for CUDA and use that as a backend if found" OFF)
-if (USE_CUDA)
-	find_package( CUDA REQUIRED )
+# Deprecated USE_CUDA option
+if(DEFINED USE_CUDA)
+  if(USE_CUDA)
+    set(ENV{HIP_PLATFORM} nvidia)
+    message(DEPRECATED "USE_CUDA is deprecated (use environment variable HIP_PLATFORM=nvidia)")
+  else()
+    set(ENV{HIP_PLATFORM} amd)
+    message(DEPRECATION "USE_CUDA is deprecated (use environment variable HIP_PLATFORM=amd)")
+  endif()
 endif()
 
 # Hip headers required of all clients; clients use hip to allocate device memory
-if( USE_CUDA )
-    find_package( hip CONFIG PATHS ${HIP_DIR} ${ROCM_PATH} /opt/rocm )
-    # support for cuda backend with hip < 6.0
-    if( NOT hip_FOUND )
-        find_package( HIP MODULE REQUIRED )
-        list( APPEND HIP_INCLUDE_DIRS "${HIP_ROOT_DIR}/include" )
-    endif()
-else()
-    find_package( hip REQUIRED CONFIG PATHS ${HIP_DIR} ${ROCM_PATH} /opt/rocm)
+find_package( hip CONFIG PATHS ${HIP_DIR} ${ROCM_PATH} /opt/rocm )
+
+# support for cuda backend with hip < 6.0
```

Comment:
This will work for old version of HIP as long as the user doesn't have HIP for the AMD platform installed (as otherwise it might have been found but not work). I'm not sure if it's worth working around that issue. Up to you.